firmware/boron-ins-fsm/src/consoleFunctions.cpp

```diff
@@ -221,56 +221,63 @@ int ins_threshold_set(String input){
+    EEPROM.get(ADDR_IM_DOORID, doorIDHolder.byte1);  
+    EEPROM.get((ADDR_IM_DOORID + 1), doorIDHolder.byte2);  
+    EEPROM.get((ADDR_IM_DOORID + 2), doorIDHolder.byte3);  
 
-    char buffer[64];
     snprintf(buffer, sizeof(buffer), "{\"byte1\":\"%02X\", \"byte2\":\"%02X\", \"byte3\":\"%02X\"}", 
-            holder.byte1,holder.byte2,holder.byte3); 
-    Particle.publish("Current Door Sensor ID: ",buffer, PRIVATE);
+            doorIDHolder.byte1, doorIDHolder.byte2, doorIDHolder.byte3); 
+    Particle.publish("Current Door Sensor ID: ", buffer, PRIVATE);
+
+    // put door ID in buffer for return value
+    snprintf(buffer, sizeof(buffer), "%02X%02X%02X", doorIDHolder.byte1, doorIDHolder.byte2, doorIDHolder.byte3);
   } 
-  else //else not echo, so we have a new door ID to parse
+  else // else not echo, so we have a new door ID to parse
   {
-    //parse input string and update global door ID 
-    const char* byteholder1;
-    const char* byteholder2;
-    const char* byteholder3;
-    int split1 = command.indexOf(',');
-    byteholder1 = command.substring(0,split1).c_str();
-    globalDoorID.byte3 = (uint8_t)strtol(byteholder1,NULL,16);
-    int split2 = command.indexOf(',', split1+1);
-    byteholder2 = command.substring(split1+1,split2).c_str();
-    globalDoorID.byte2 = (uint8_t)strtol(byteholder2,NULL,16);
-    int split3 = command.indexOf(',', split2+1);
-    byteholder3 = command.substring(split2+1,split3).c_str();
-    globalDoorID.byte1 = (uint8_t)strtol(byteholder3,NULL,16);
+    // parse input string and update global door ID 
+    char* byteholder1;
+    char* byteholder2;
+    char* byteholder3;
+
+    int split1 = command.indexOf(',');                                // get index of first comma to delimit input
+    byteholder1 = strdup(command.substring(0, split1).c_str());       // get first byte of input and copy to holder variable
+    globalDoorID.byte3 = (uint8_t)strtol(byteholder1, NULL, 16);      // convert it to hex and set the third byte of the door ID
+
+    int split2 = command.indexOf(',', split1 + 1);
+    byteholder2 = strdup(command.substring(split1 + 1, split2).c_str());
+    globalDoorID.byte2 = (uint8_t)strtol(byteholder2, NULL, 16);
+
+    int split3 = command.indexOf(',', split2 + 1);
+    byteholder3 = strdup(command.substring(split2 + 1, split3).c_str());
+    globalDoorID.byte1 = (uint8_t)strtol(byteholder3, NULL, 16);
 
     //write new global door ID to flash
-    EEPROM.put(ADDR_IM_DOORID,globalDoorID.byte1);  
-    EEPROM.put((ADDR_IM_DOORID+1),globalDoorID.byte2);  
-    EEPROM.put((ADDR_IM_DOORID+2),globalDoorID.byte3);  
-  
-  } //end if-else
+    EEPROM.put(ADDR_IM_DOORID, globalDoorID.byte1);
+    EEPROM.put((ADDR_IM_DOORID + 1), globalDoorID.byte2);
+    EEPROM.put((ADDR_IM_DOORID + 2), globalDoorID.byte3);
 
-  return 1;
+    // put door ID in buffer for return value
+    snprintf(buffer, sizeof(buffer), "%02X%02X%02X", globalDoorID.byte1, globalDoorID.byte2, globalDoorID.byte3);
```

Comment:
Here and for the `e` echo return value, I was actually expecting this to return with the bytes in the opposite order:
`snprintf(buffer, sizeof(buffer), "%02X%02X%02X", globalDoorID.byte3, globalDoorID.byte2, globalDoorID.byte1);`. This way it will match the order on the door sensor sticker itself, right?

Reply:
This is completely true, and I've changed it in my latest commit. I also renamed doorIDHolder to use the same byte order as globalDoorID as I think that is what caused the confusion